lib/services/watcher.go

```diff
@@ -729,8 +729,13 @@ func (p *databaseCollector) getResourcesAndUpdateCurrent(ctx context.Context) er
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	p.current = newCurrent
-	p.DatabasesC <- databases
-	return nil
+
+	select {
+	case <-ctx.Done():
+		return trace.Wrap(ctx.Err())
+	case p.DatabasesC <- databases:
+		return nil
+	}
```

Comment:
Always nice to have a return at the end of a function when possible.
```suggestion
	case p.DatabasesC <- databases:
	}
	return nil
```